### ISSUE TITLE
Issues accessing repositories prevents retrieving snapshots from `sys.snapshots`

### ISSUE THREAD

**Issue by SStorm** [crate/crate#13743]
### CrateDB version

5.2.2, 5.2.6

### CrateDB setup information

Standard CrateDB Cloud cluster

### Problem description

If there is more than one repository configured in a cluster, and at least one of them has access issues (i.e. no longer valid credentials), the query to retrieve snapshots (`SELECT * FROM sys.snapshots`) becomes inconsistent and sometimes returns no results.

### Steps to Reproduce

1. Create two repositories
2. Perform some snapshots
3. Break auth to one of them

### Actual Result

`SELECT * FROM sys.snapshots` becomes inconsistent

### Expected Result

One of the repositories being unavailable does not impact retrieval of snapshots from others.

**Comment by jeeminso** [comment]
Thanks for reporting @SStorm. I was not able to reproduce this using `s3` and `fs` locally. Could you give more specific scenario?
I have noticed that deactivating s3 access_key took longer time to be reflected on `select * from sys.snapshots` but hard to relate it to this bug.

**Comment by SStorm** [comment]
Info provided privately.

**Comment by jeeminso** [comment]
Thank you @SStorm for the details. The credential issue was not required to observe the inconsistent results from `SELECT * FROM sys.snapshots`.

**Comment by jeeminso** [comment]
It looks like the inconsistency is due to denied accesses from S3:

```
cause = {RepositoryException@20451} "org.elasticsearch.repositories.RepositoryException: [system_backup_20221215152410] Unexpected exception when loading repository data"
 repository = "system_backup_20221215152410"
 metadata = {HashMap@20457}  size = 0
 headers = {HashMap@20458}  size = 0
 backtrace = {Object[7]@20459} 
 detailMessage = "[system_backup_20221215152410] Unexpected exception when loading repository data"
 ```
 ```
 cause = {AmazonS3Exception@20461} "com.amazonaws.services.s3.model.AmazonS3Exception: Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied;   cause = {AmazonS3Exception@20461} "com.amazonaws.services.s3.model.AmazonS3Exception: Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: T32JSAV86VB0BV2W; 
stackTrace = {StackTraceElement[23]@20506} 
 0 = {StackTraceElement@20529} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1879)"
 1 = {StackTraceElement@20530} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1418)"
 2 = {StackTraceElement@20531} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1387)"
 3 = {StackTraceElement@20532} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1157)"
 4 = {StackTraceElement@20533} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:814)"
 5 = {StackTraceElement@20534} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:781)"
 6 = {StackTraceElement@20535} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:755)"
 7 = {StackTraceElement@20536} "com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:715)"
 8 = {StackTraceElement@20537} "com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:697)"
 9 = {StackTraceElement@20538} "com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:561)"
 10 = {StackTraceElement@20539} "com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:541)"
 11 = {StackTraceElement@20540} "com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5456)"
 12 = {StackTraceElement@20541} "com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5403)"
 13 = {StackTraceElement@20542} "com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1524)"
 14 = {StackTraceElement@20543} "com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1378)"
 15 = {StackTraceElement@20544} "org.elasticsearch.repositories.s3.S3BlobContainer.readBlob(S3BlobContainer.java:101)"
 16 = {StackTraceElement@20545} "org.elasticsearch.repositories.blobstore.BlobStoreRepository.getRepositoryData(BlobStoreRepository.java:1333)"
 17 = {StackTraceElement@20546} "org.elasticsearch.repositories.blobstore.BlobStoreRepository.doGetRepositoryData(BlobStoreRepository.java:1248)"
 18 = {StackTraceElement@20547} "org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)"
 19 = {StackTraceElement@20548} "org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)"
 20 = {StackTraceElement@20549} "java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)"
 21 = {StackTraceElement@20550} "java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)"
 22 = {StackTraceElement@20551} "java.base/java.lang.Thread.run(Thread.java:1589)"
```

I will continue to investigate why it is getting access denied (and cannot reproduce it locally).

**Comment by jeeminso** [comment]
Thank you for reporting and sorry for late updates, but I must have shared most of these already.
I experimented on the culprit cluster with my AWS S3 buckets that several successive invocations of `select repository, count(*) from sys.snapshots group by (repository)` returned as expected. I still do not know why some of the S3 buckets are getting intermittent access denied errors but since it did not impact my snapshots, `One of the repositories being unavailable does not impact retrieval of snapshots from others.` holds true.
We have agreed that silently returning partial results from `select * from sys.snapshots` seems like a bug but on a second thought, `select * from sys.snapshots` did not return partial results but it returned all snapshots from all repos reachable at that point in time.

**Comment by SStorm** [comment]
I have to insist that this is still a problem and is impacting us. Adding more than one repository will lead to non-deterministic behaviour when querying snapshots. Happy to share more details privately.

**Comment by seut** [comment]
@SStorm 
Thank you for insisting on the issue, it was indeed a bug. 
It has been fixed and will be available with the next 5.3 hotfix release.
